modular_skyrat/modules/xenoarch/code/modules/research/xenoarch/strange_rock.dm

```diff
@@ -0,0 +1,212 @@
+/obj/item/xenoarch/strange_rock
+	name = "strange rock"
+	desc = "A mysterious, strange rock that has the potential to have a wonderful item. Also possible for it to have our disposed garbage."
+	icon_state = "rock"
+
+	var/max_depth //the max depth
+	var/safe_depth //the depth away from the max depth
+	var/item_depth //the depth between the max and the safe, containing the item
+	var/dug_depth = 0 //the depth that has been currently dug
+
+	var/hidden_item //the item that is held
+
+	var/measured = FALSE //if the item has been tape-measured
+	var/scanned = FALSE //if the item has been scanned
```

Comment:
Documentation on variables should be done like this so that they appear above the variables when you hover over them using the appropriate VSC DM extension. Also, for AUTODOC, I believe.
```suggestion
	/// The max depth of the strange rock.
	var/max_depth
	/// The depth away from the max depth.
	var/safe_depth
	/// The depth between the max and the safe depths, containing the item.
	var/item_depth
	/// The depth that has been dug so far.
	var/dug_depth = 0

	/// The item the strange rock contains.
	var/hidden_item

	/// If the strange rock has been tape-measured.
	var/measured = FALSE
	/// If the strange rock has been scanned.
	var/scanned = FALSE
```